src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -391,18 +391,31 @@ public override void VisitMethod(IMethodSymbol symbol)
                     var name = format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.UseMetadataMethodNames) || symbol.ContainingType == null || symbol.ContainingType.IsAnonymousType
                         ? symbol.Name
                         : symbol.ContainingType.Name;
-                    builder.Add(CreatePart(SymbolDisplayPartKind.MethodName, symbol, name));
+
+                    // In the case that symbol.containingType is null (which should never be the case here) we will fallback to the MethodName symbol part
+                    if (!TryGetPartKindForContainingType(symbol, out var displayPartKind))
+                    {
+                        displayPartKind = SymbolDisplayPartKind.MethodName;
+                    }
+
+                    builder.Add(CreatePart(displayPartKind, symbol, name));
                     break;
                 case MethodKind.Destructor:
+                    // In the case that symbol.containingType is null (which should never be the case here) we will fallback to the MethodName symbol part
+                    if (!TryGetPartKindForContainingType(symbol, out var partKind))
+                    {
+                        partKind = SymbolDisplayPartKind.MethodName;
+                    }
+
```

Comment:
Consider adding a scope for each of these `case` blocks so we can use the same `partKind` variable name in both.